### PR TITLE
feat: add auto-complete suggestions

### PR DESCRIPTION
**Pull request by ctron**

### CODE REVIEW

**Comment by ctron** [comment]
[preview.webm](https://github.com/trustification/trustification/assets/202474/28735aa4-3d50-4a60-8799-7197f836fbad)

File: spog/api/src/endpoints/suggestion.rs
Comment (carlosthe19916) [inline spog/api/src/endpoints/suggestion.rs:50]: Currently the "suggestions" search is applied only to "packages (v11y)". I just want to make sure we can extend the "suggestions" also to "sboms" and "packages". The limitation I see (i might be wrong) is that each entity "cve", "sbom", and "packages" are not related and don't share a single index where we can query the "suggestions".

- The final objective might have a single search endpoint and get suggestions for all entities "cve", "sbom", and "packages". How easy or difficult it might be to extend the current code to include those 3 entities in the future? I'm not saying we have to do it now, we can start small and incrementally enhance things.

**Review by carlosthe19916 — COMMENTED** [review]
It might be premature to ask but having the suggestions also in the following page might be helpful (to be consistent with the search input in the home page)

Screenshot from 2023-12-11 09-55-15

File: spog/api/src/endpoints/suggestion.rs
Comment (ctron) [reply spog/api/src/endpoints/suggestion.rs:50]: Absolutely. The one of the next steps I want to take a look at is to add at least SBOMs. Advisories might be a bit trickier. Not due to technical difficulties, but due to aggregating the results. Otherwise we might have two CVEs with the same ID (one from raw CVE, the other from a VEX with the same ID). So it's more about defining what we want.

**Comment by ctron** [comment]
> It might be premature to ask but having the suggestions also in the following page might be helpful (to be consistent with the search input in the home page)

On Friday, yes. On Monday, no! :) So yes, it should be there in the end.

**Review by carlosthe19916 — APPROVED** [review]
@ctron This looks good to me. Just fix the test that is failing (some clippy thing). And once we have the approval from @desmax74 for merging PRs without any risk we can merge this PR

**Comment by ctron** [comment]
I've got one more problem to sort out. On the search page the auto-complete opens up by default on the first render. Otherwise, yes, I think it's good to go.

**Comment by ctron** [comment]
@carlosthe19916 I think it's good now.

**Comment by carlosthe19916** [comment]
Awesome! PR approved and on the queue for when we are able to merge PRs

**Pull request merged**
